### DISCUSSION TITLE
branch vs fork to explor some future enhancements?

### DISCUSSION THREAD

**Discussion by stefankinell** [karttapullautin/karttapullautin#24] [General]
As we discussed this spring I have some ideas of future enhancements of the script (or really things I would need to shorten processing time when handling large amounts of laz-files) to e.g. enable filtering of input files and run only parts of the script like to enable batch processing for vegetation only. 

Do you have any thoughts of what the best way to do this without messing up the core project, but still save it so it can be encorporated later on in main release? I assume the choices I have are either to fork it, or do a branch. I assume a branch would be smarter so I can enjoy other enhancements that are done. Without to mouch thoughts going into it I assume the changes I want to do will be done in the main perl script and perhaps the .ini-file

**Comment by rphlo** [comment]
The way github works is that to do any change to a project you dont own, you must first fork it. Then it is up to you to create branches in your repo to tests stuff. Once you think you reached a point where your changes work and can be beneficial for others, then you can submit a pull request from you branch to this repo.

**Reply by stefankinell** [reply]
Thanks, then I learnt something new today. I will the fork it and play around. If it turns out well I will try to submit it to the main project.
